crates/api/src/timestamp.rs

```diff
@@ -0,0 +1,69 @@
+/// Kitsune2 timestamp.
+///
+/// Internally i64 microseconds from unix epoch.
+// Legacy kitsune had a massively overcomplicated timestamp type.
+// - We don't need chrono
+// - We don't need human readable times
+// - We don't need sqlite transformation functions
+// - We DO need the ability to get the current "now" timestamp
+#[derive(
+    Debug,
+    Clone,
+    Copy,
+    PartialEq,
+    Eq,
+    PartialOrd,
+    Ord,
+    Hash,
+    serde::Serialize,
+    serde::Deserialize,
+)]
+#[serde(transparent)]
+pub struct Timestamp(i64);
+
+impl Timestamp {
+    /// Construct a new timestamp of "now".
+    pub fn now() -> Self {
+        Self(
+            std::time::SystemTime::now()
+                .duration_since(std::time::SystemTime::UNIX_EPOCH)
+                .expect("invalid system clock")
```

Comment:
This doesn't happen on functional machines, right?!

Reply:
It might be possible to hack your OS to be at a time before the epoch... but I imagine so much other software would crash that the system would essentially be unusable.